Test/QuickCheck/Arbitrary.hs

```diff
@@ -432,9 +435,15 @@ instance Arbitrary a => Arbitrary [a] where
   shrink = shrink1
 
 #ifndef NO_NONEMPTY
+instance Arbitrary1 NonEmpty where
+  liftArbitrary arb = liftM2 (:|) arb (liftArbitrary arb)
+  liftShrink shr (x :| xs) =
+      [ x' :| xs | x' <- shr x ] ++
+      [ x :| xs' | xs' <- liftShrink shr xs ]
+
```

Comment:
This is a bit worse than before because it won't try removing the first element from the list. We should probably just keep the old shrinking approach (it adds very little overhead).